### PR TITLE
HOTT-1252 Check the correctness of sequence for the CDS and Taric updates.

### PR DESCRIPTION
**Pull request by alexdesi**
### Jira link

[HOTT-1252](https://transformuk.atlassian.net/browse/HOTT-1252)

### What?
We want to identify whether the current update is the next increment for the sequence based on its filename. In case it is not the expected update in sequence, it should not be run, and an alert must be sent to an appropriate channel.

### Why?
To avoid data inconsistency.

### CODE REVIEW

File: app/lib/tariff_synchronizer.rb [hunk 1]
```diff
@@ -264,6 +269,35 @@ def initial_update_date_for(update_type)
 
   private
 
+  def correct_sequence_for_pending_cds_file?
+    last_pending = TariffSynchronizer::CdsUpdate.last_pending&.filename
```
[the remark this answers is not in the document]
Comment (willfish) [reply app/lib/tariff_synchronizer.rb:273]:
```ruby
    raise FailedUpdatesError unless TariffSynchronizer::CdsUpdate.correct_sequence_for_pending_cds_file?
```

File: app/lib/tariff_synchronizer.rb [hunk 2]
```diff
@@ -264,6 +269,35 @@ def initial_update_date_for(update_type)
 
   private
 
+  def correct_sequence_for_pending_cds_file?
```
Comment (willfish) [inline app/lib/tariff_synchronizer.rb:272]:
```suggestion
  def check_next_file_in_sequence
```

File: app/lib/tariff_synchronizer.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexdesi) [reply app/lib/tariff_synchronizer.rb:273]: and maybe the
`match(/^tariff_dailyExtract_v1_(?<year>\d{4})(?<month>\d{2})(?<day>\d{2})T\d+\.gzip$/)`
should be moved in the `CdsUpdate` too.

I was thinking to add 
`def last_pending_file_date`
there.

File: app/lib/tariff_synchronizer.rb [hunk 3]
```diff
@@ -127,6 +129,7 @@ def apply(reindex_all_indexes: false)
 
   def apply_cds(reindex_all_indexes: false)
     check_tariff_updates_failures
+    check_sequence_for_pending_cds_file
```
Comment (alexdesi) [inline app/lib/tariff_synchronizer.rb:132]: now this follows the same pattern as `check_tariff_updates_failures`

**Review by willfish — APPROVED** [review]

**Pull request merged**